src/Confluent.Kafka/Deserializers.cs

```diff
@@ -0,0 +1,279 @@
+
+        /// <summary>
+        ///     Deserializes a big endian encoded (network byte ordered) <see cref="System.Int64"/> value from a byte array.
+        /// </summary>
+        /// <param name="data">
+        ///     A byte array containing the serialized <see cref="System.Int64"/> value (big endian encoding)
+        /// </param>
+        /// <param name="topic">
+        ///     The topic associated with the data (ignored by this deserializer).
+        /// </param>
+        /// <param name="isNull">
+        ///     True if the data is null, false otherwise.
+        /// </param>
+        /// <returns>
+        ///     The deserialized <see cref="System.Int64"/> value.
+        /// </returns>
+        public static long Long(string topic, ReadOnlySpan<byte> data, bool isNull)
+        {
+            if (isNull)
+            {
+                throw new ArgumentException($"Arg [{nameof(data)}] is null");
+            }
+
+            if (data.Length != 8)
+            {
+                throw new ArgumentException($"Size of {nameof(data)} received by Deserializer<Long> is not 8");
+            }
+
+            // network byte order -> big endian -> most significant byte in the smallest address.
+            long result = ((long)data[0]) << 56 |
+                ((long)(data[1])) << 48 |
+                ((long)(data[2])) << 40 |
+                ((long)(data[3])) << 32 |
+                ((long)(data[4])) << 24 |
+                ((long)(data[5])) << 16 |
+                ((long)(data[6])) << 8 |
+                (data[7]);
+            return result;
+        }
+
+        /// <summary>
+        ///     Deserializes a big endian encoded (network byte ordered) <see cref="System.Int32"/> value from a byte array.
+        /// </summary>
+        /// <param name="data">
+        ///     A byte array containing the serialized <see cref="System.Int32"/> value (big endian encoding).
+        /// </param>
+        /// <param name="topic">
+        ///     The topic associated with the data (ignored by this deserializer).
+        /// </param>
+        /// <param name="isNull">
+        ///     True if the data is null, false otherwise.
+        /// </param>
+        /// <returns>
+        ///     The deserialized <see cref="System.Int32"/> value.
+        /// </returns>
+        public static int Int32(string topic, ReadOnlySpan<byte> data, bool isNull)
+        {
+            if (isNull)
+            {
+                throw new ArgumentNullException($"Arg {nameof(data)} is null");
+            }
+
+            // network byte order -> big endian -> most significant byte in the smallest address.
+            return
+                (((int)data[0]) << 24) |
+                (((int)data[1]) << 16) |
+                (((int)data[2]) << 8) |
+                (int)data[3];
+        }
+
+        /// <summary>
+        ///     Deserializes a big endian encoded (network byte ordered) System.Single value from a byte array.
```

Comment:
Is this IEEE-754 compatible? It really should be

Reply:
most likely. it's the same as previous serializers. I'll add it as a todo to #614 to double check.